backend/telemetry/src/main/java/org/sonarsource/sonarlint/core/telemetry/TelemetryManager.java

```diff
@@ -40,8 +45,22 @@ public class TelemetryManager {
   }
 
   void enable(TelemetryLiveAttributes telemetryLiveAttributes) {
-    storageManager.tryUpdateAtomically(data -> data.setEnabled(true));
-    uploadLazily(telemetryLiveAttributes);
+    storageManager.tryUpdateAtomically(localStorage -> {
+      localStorage.setEnabled(true);
+      if (isGracePeriodElapsed(localStorage.lastUploadTime())) {
+        uploadAndClearTelemetry(telemetryLiveAttributes, localStorage);
+      }
+    });
+  }
+
+  private static boolean isGracePeriodElapsed(LocalDateTime lastUploadTime) {
```

Comment:
I really like the `isGracePeriodElapsed` method name as it actually reflects what it's doing as opposed to the `dayChanged`. 
I know it's a years-old thing not introduced in this PR but I wonder whether to rename the `dayChanged` to `isGracePeriodElapsedAndDayChanged` or similar and use directly that,

Reply:
I renamed both the `dayChanged` utility method and the private method to what you suggested. I kept the instance method in order to use one fewer parameter.